### PR TITLE
Allow group admins to survey their own events without needing an RSVP

### PR DESCRIPTION
**Pull request by kdeloach**
This change allows group admins to survey their own events without
taking up an RSVP slot. Clicking the Treecorder link in the nav while
an event is in progress should take the group admin to the event
check-in page.

To test:
1. Login as a group admin with an event in progress (don't RSVP)
2. Click the Treecorder link
3. You should be taken to the "survey for event" page

Connects #1692

### CODE REVIEW

File: src/nyc_trees/apps/event/helpers.py [hunk 1]
```diff
@@ -2,17 +2,21 @@
 
 
 def user_is_rsvped_for_event(user, event):
-    # The filter uses user.id so that the user argument can be either
-    # a real User or the anonymous user (not logged in). The anonymous
-    # user is a SimpleLazyObject, which you cannot use in a filter.
-    return EventRegistration.objects\
```
Comment (maurizi) [inline src/nyc_trees/apps/event/helpers.py]: This comment and the fix it references (using `user.id` instead of just `user` fixed a bug we encountered several times).  I would like us to keep it in place.

File: src/nyc_trees/apps/event/models.py [hunk 2]
```diff
@@ -231,6 +231,16 @@ def my_events_now(cls, user):
...
         attended = [r.event for r in registrations if r.did_attend]
         non_attended = [r.event for r in registrations if not r.did_attend]
+
+        # Group admins don't need to RSVP to be able to survey events.
+        # These filters are copy & pasted from the query directly above.
+        admin_events = Event.objects \
+            .filter(group__admin=user, begins_at__gte=beginning_of_day_utc) \
+            .filter(Q(ends_at__lte=end_of_day_utc) |
+                    Q(begins_at__lte=now_utc + STARTING_SOON_WINDOW,
+                      ends_at__gt=now_utc))
```
Comment (maurizi) [inline src/nyc_trees/apps/event/models.py]: No need to do this now if you don't want to, but FYI a nice way to avoid this duplication is to do something like this:

```python
events_soon = Event.objects \
    .filter(begins_at__gte=beginning_of_day_utc) \
    .filter(Q(ends_at__lte=end_of_day_utc) |
            Q(begins_at__lte=now_utc + STARTING_SOON_WINDOW,
              ends_at__gt=now_utc))

registrations = EventRegistration.objects \
    .filter(user=user) \
    .filter(event_id__in=events_soon.values_list('id', flat=True))

attended = [r.event for r in registrations if r.did_attend]
non_attended = [r.event for r in registrations if not r.did_attend]

attended = attended + list(events_soon.filter(group__admin=user))
```

**Comment by maurizi** [comment]
@azavea-bot rebuild

**Comment by maurizi** [comment]
As the group admin, I can click the "Attending" button on the group detail page.  It will still say "Attending", but it makes the "Check-in guests" button appear/disappear.  We should probably just disable clicking on it if you are the group admin.

**Comment by maurizi** [comment]
Events alerts for groups that you are the admin of do not appear in the header on the dashboard, and do not appear in the "Attending" section of the event list on the dashboard.

The event alert seems important since it is one of the easiest ways for group admins to be able to check users into the event.

The event not showing in the event list seems less important, but it is confusing, since if you go to the event detail, you are shown as "Attending"

File: src/nyc_trees/apps/event/helpers.py (hunk 1 above)
Comment (kdeloach) [inline src/nyc_trees/apps/event/helpers.py]: I decided it's not worth making a query if we already know the answer.

File: src/nyc_trees/apps/event/models.py (hunk 2 above)
Comment (kdeloach) [inline src/nyc_trees/apps/event/models.py]: OK I like this better, will update.

Comment (kdeloach) [inline src/nyc_trees/apps/event/models.py]: Assuming we can safely drop the `did_attend=True` criteria from this query.

**Comment by kdeloach** [comment]
Added a commit to make the RSVP button static for group admins and also fixed event alerts not appearing on the dashboard.

**Comment by kdeloach** [comment]
Added another commit to display events on the dashboard and event list.

**Comment by kdeloach** [comment]
Pushed a fix to the event ordering in `next_event_starting_soon`. 
We can't actually do sorting when combining the results because the event begins_at field is no longer in context at that point (it's just a list of RSVPs). I moved the admin "RSVPs" to the beginning of the list so they take precedence over other events, and hopefully that will be good enough.

**Comment by maurizi** [comment]
+1, works well.

I actually see *two* choices in the treecorder modal when I click on the treecorder link from the nav, one to "check in" which takes me to the admin check-in page, and one to "map" which takes me to the survey page, but that seems fine.

**Comment by maurizi** [comment]
Though it looks like your unit test is failing

**Comment by kdeloach** [comment]
You actually found an edge case I didn't think about. It only happens if a group admin RSVPs to their own event. However, group admins no longer have the ability to RSVP to their own events, so this problem should not happen in production.

Now that I think about it... maybe we should link group admins to the check-in page instead of directly to the survey page?

**Comment by kdeloach** [comment]
Fixed unit test.

**Comment by kdeloach** [comment]
Fixed the edge case where, if the group admin RSVPs and checks into their own events, it would display 2 links in the Treecorder urls panel. Now it should always display the check-in page for group admins.

**Comment by maurizi** [comment]
But now your unit test is broken again :(

**Comment by kdeloach** [comment]
Yup, I can see that.
image

**Comment by kdeloach** [comment]
Fixed unit tests.

**Comment by maurizi** [comment]
+1, works great
